services/api-server/src/simcore_service_api_server/core/health_checker.py

```diff
@@ -0,0 +1,120 @@
+from prometheus_client import CollectorRegistry, Gauge
+from pydantic import NonNegativeInt, PositiveFloat, PositiveInt
+from servicelib.background_task import start_periodic_task, stop_periodic_task
+from servicelib.fastapi.dependencies import get_app
+from servicelib.rabbitmq import RabbitMQClient
+from simcore_service_api_server.models.schemas.jobs import JobID, JobLog
+from simcore_service_api_server.services.log_streaming import LogDistributor
+
+from .._meta import PROJECT_NAME
+
+METRICS_NAMESPACE: Final[str] = PROJECT_NAME.replace("-", "_")
+
+_logger = logging.getLogger(__name__)
+
+
+class ApiServerHealthChecker:
+    def __init__(
+        self,
+        *,
+        registry: CollectorRegistry,
+        log_distributor: LogDistributor,
+        rabbit_client: RabbitMQClient,
+        timeout_seconds: PositiveFloat,
+        allowed_health_check_failures: PositiveInt,
+    ) -> None:
+        self._registry = registry
+        self._log_distributor: LogDistributor = log_distributor
+        self._rabbit_client: RabbitMQClient = rabbit_client
+        self._timeout_seconds = timeout_seconds
+        self._allowed_health_check_failures = allowed_health_check_failures
+
+        self._logstreaming_queues_gauge = Gauge(
+            "log_stream_queue_length",
+            "#Logs in log streaming queue",
+            ["job_id"],
+            namespace=METRICS_NAMESPACE,
+        )
+        self._health_check_qauge = Gauge(
+            "log_stream_health_check",
+            "#Failures of log stream health check",
+            namespace=METRICS_NAMESPACE,
+        )
+        self._health_check_failure_count: NonNegativeInt = 0
+        self._dummy_job_id: JobID = uuid4()
+        self._dummy_queue: asyncio.Queue[JobLog] = asyncio.Queue()
+        self._dummy_message = LoggerRabbitMessage(
+            user_id=UserID(123456789),
+            project_id=self._dummy_job_id,
+            node_id=uuid4(),
+            messages=["Api-server health check message"],
+        )
+        self._background_task: asyncio.Task | None = None
+        _logger.info("Api server health check dummy job_id=%s", f"{self._dummy_job_id}")
+
+    async def setup(self, health_check_task_period_seconds: PositiveFloat):
+        await self._log_distributor.register(
+            job_id=self._dummy_job_id, queue=self._dummy_queue
+        )
+        self._background_task = start_periodic_task(
+            task=self._background_task_method,
+            interval=timedelta(seconds=health_check_task_period_seconds),
+            task_name="api_server_health_check_task",
+        )
+
+    async def teardown(self, timeout_seconds: PositiveFloat):
+        if self._background_task:
+            await stop_periodic_task(self._background_task, timeout=timeout_seconds)
+        await self._log_distributor.deregister(job_id=self._dummy_job_id)
+
+    @property
+    def healthy(self) -> bool:
+        return self._health_check_failure_count <= self._allowed_health_check_failures
```

Comment:
The rabbitmq client also has a healthy property, not sure you make use of it. This is used in several different locations, and maybe if it is enough or can be used as well or in conjonction.
question, did you also check that?
https://github.com/ITISFoundation/osparc-simcore/blob/12310f80ac388c4de8118f9059bc6c4412ab22ed/services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/api/rest/_health.py

Reply:
Added, thanks for reminding me.